pkg/nsx/services/vpc/vpc.go

```diff
@@ -471,3 +523,236 @@ func (s *VPCService) Cleanup() error {
 
 	return nil
 }
+
+func (service *VPCService) needUpdateRule(rule *model.Rule, externalCIDR []string) bool {
+	des := rule.DestinationGroups
+	currentDesSet := sets.Set[string]{}
+	for _, group := range des {
+		currentDesSet.Insert(group)
+	}
+	if len(externalCIDR) != len(currentDesSet) {
+		return true
+	}
+	for _, newGroup := range externalCIDR {
+		if !currentDesSet.Has(newGroup) {
+			return true
+		}
+	}
+	return false
+}
+
+func (service *VPCService) getExternalIpBlock(vpcName string, namespace string) ([]string, error) {
+	// read corresponding vpc network config from store
+	nc_name, err := service.getNetworkconfigNameFromNS(namespace)
+	if err != nil {
+		log.Error(err, "failed to get network config name for VPC when creating NSX VPC rule", "VPC", vpcName, "namespace", namespace)
+		return nil, err
+	}
+	nc, _exist := service.GetVPCNetworkConfig(nc_name)
+	if !_exist {
+		log.Error(err, "failed to read network config  when creating NSX VPC rule", "nc_name", nc_name)
+		return nil, err
+	}
+	log.Info("get external ip block", "ip block", nc.ExternalIPv4Blocks)
+	return nc.ExternalIPv4Blocks, nil
+
+}
+
+func (service *VPCService) getIpblockCidr(cidrID []string) (result []string, err error) {
+	for _, cidr := range cidrID {
+		ipblock := service.PubIpblockStore.GetByKey(cidr)
+		if ipblock == nil {
+			err = errors.New("ipblock not found")
+			log.Error(err, "failed to get public ipblock", "path", cidr)
+			return
+		} else {
+			result = append(result, *ipblock.Cidr)
+		}
+	}
+	return
+}
+
+func (service *VPCService) CreateOrUpdateAVISecurityPolicy(vpc *model.Vpc, namespace string) error {
+	if !service.supportAviAllowRule {
+		return nil
+	}
+	path := strings.Split(*vpc.Path, "/")
```

Comment:
we already a function to parse vpc info,
https://github.com/vmware-tanzu/nsx-operator/blob/vpc_dev/pkg/nsx/services/common/builder.go#L40
we can use this function to do it

Reply:
done